app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1936,4 +1937,145 @@ class PatientRepositoryAndroidTest {
+    val searchResultsWithLastPartMatching = searchResults("67890")
+    val searchResultsWithMiddlePartMatching = searchResults("45678")
+    val searchResultsWithNoPartMatching = searchResults("22222")
+
+    // then
+    assertThat(searchResultsWithExactMatch)
+        .containsExactly(patientWithExactlyMatchingNumber)
+    assertThat(searchResultsWithFirstPartMatching)
+        .containsExactly(
+            patientWithFirstPartMatchingNumber,
+            patientWithExactlyMatchingNumber
+        )
+    assertThat(searchResultsWithLastPartMatching)
+        .containsExactly(
+            patientWithLastPartMatchingNumber,
+            patientWithExactlyMatchingNumber
+        )
+    assertThat(searchResultsWithMiddlePartMatching)
+        .containsExactly(
+            patientWithMiddlePartMatchingNumber,
+            patientWithExactlyMatchingNumber
+        )
+    assertThat(searchResultsWithNoPartMatching)
+        .isEmpty()
+  }
+
+  @Test
+  fun searching_for_a_patient_by_phone_number_must_return_results_ordered_by_patient_name() {
+    fun createPatientWithPhoneNumber(
+        patientName: String,
+        phoneNumber: String
+    ) {
+
+      val patientProfile = testData
+          .patientProfile(patientUuid = UUID.randomUUID(), generatePhoneNumber = !phoneNumber.isBlank())
+          .let { patientProfile ->
+            patientProfile.copy(
+                patient = patientProfile.patient.copy(fullName = patientName),
+                phoneNumbers = patientProfile
+                    .phoneNumbers
+                    .take(1)
+                    .map { it.copy(number = phoneNumber) }
+            )
+          }
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+    }
+
+    fun searchResults(phoneNumber: String): List<String> {
+      return patientRepository
+          .searchByPhoneNumber(phoneNumber)
+          .blockingFirst()
+          .map { it.fullName }
+    }
+
+    // given
+    createPatientWithPhoneNumber(patientName = "Abhinav", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "Chitra", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "abhinav", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "Fatima", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "esther d' souza", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "Dhruv Saxena", phoneNumber = "1234567890")
+    createPatientWithPhoneNumber(patientName = "Dhruv Acharya", phoneNumber = "1234567890")
+
+    // when
+    val searchResults = searchResults(phoneNumber = "1234567890")
+        // Converting to lowercase because of different SQLite implementations in
+        // platforms causes flaky tests because the ordering of small and capital
+        // strings which are identical is platform-dependent.
+        .map { it.toLowerCase(Locale.ENGLISH) }
```

Comment:
Just a doubt.
Does this mean that the search SQL query can return different results based on the platform?

Reply:
The default ordering of results in SQL is undefined. However, the part which I am talking about is if we have cases like this where we explicitly order by name and ignore the case and there are names which are exactly the same but in different cases (like "Abhinav" and "abhinav").

Different runs of the test will sometimes return ["Abhinav", "abhinav"] and sometimes ["abhinav", "Abhinav"] and causes flakiness. So I explicitly lowercased the results in the tests.

Reply:
And we don't care about this order to be the same everywhere?

Reply:
We can't force it. I think it's ok for a user to see ["Abhinav", "abhinav"] in one run and ["Abhinav, "abhinav"] in another, because the solution to make sure it always remains the same order is a lot more effort than the benefit, IMO.